src/v/coproc/offset_storage_utils.cc

```diff
@@ -116,9 +52,11 @@ ss::future<ntp_context_cache> recover_offsets(
     /// Query the total size, and read the rest of the data
     const size_t size = co_await reader.get_snapshot_size();
     iobuf data = co_await read_iobuf_exactly(reader.input(), size);
+    iobuf_parser iobuf_p(std::move(data));
     co_await reader.close();
     /// Deserialize the data
-    iresults_map irm = co_await deserialize_data_field(std::move(data));
+    iresults_map irm = co_await reflection::async_adl<iresults_map>{}.from(
+      iobuf_p);
```

Comment:
are there backwards incompatible issues with any of these changes to async adl?

Reply:
No, async adl is just an interface and the logic of serializing and deserializing underlying lists and maps was duplicated.

Reply:
is that really true? it seems that the serde for at least one case was done by hand

https://github.com/vectorizedio/redpanda/pull/1427/files#diff-fa39d4bcec986f12a354a9f4141f51609a856108f0394a6cc748b28c550f0a3aL47-L48

Reply:
Ah are you referring to its on disk format? In that case maybe. That source was never needed since think the default impl for adl can serialize structs, and i think it does it one field at a time in order, which would mean that both formats are identical.

Reply:
yes. if any data might be on disk make sure this still works. otherwise things go boom